x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -333,34 +333,34 @@ export const buildEqlDataProviderOrFilter = (
     return {
       dataProviders: [],
       filters: buildAlertsKqlFilter(
-        'signal.group.id',
+        'kibana.alert.group.id',
         ecs.reduce<string[]>((acc, ecsData) => {
-          const signalGroupId = ecsData.signal?.group?.id?.length
-            ? ecsData.signal?.group?.id[0]
-            : 'unknown-signal-group-id';
-          if (!acc.includes(signalGroupId)) {
-            return [...acc, signalGroupId];
+          const kibanaAlertGroupId = ecsData.kibana?.alert?.group?.id?.length
```

Comment:
We still may have legacy signals that contain `signal.group.id`, so I believe this code needs to handle both. I did something like that in https://github.com/elastic/kibana/pull/121256 ... there may be a way we can normalize this further up the chain, but it was too big an undertaking for me to tackle at the moment.

Reply:
Thanks for the feedback Madi! I made the changes as well as added a `getFieldKey` helper or maybe I can modify `getField` to return `{ key, value }` instead?